Non-unified build fixes, early October 2023 edition

donny-dont opened the pull request:
#### 4e458d4e5c4b3b4b84a4fc9af6429dd708214038
<pre>
Non-unified build fixes, early October 2023 edition
<a href="https://bugs.webkit.org/show_bug.cgi?id=262719">https://bugs.webkit.org/show_bug.cgi?id=262719</a>

Unreviewed non-unified build fix.

Run a non-unified build of Windows with clang-cl and fix any compilation
issues and warnings. Warnings were all around unused code.

* Source/JavaScriptCore/bytecode/AccessCase.cpp:
* Source/JavaScriptCore/bytecode/GetterSetterAccessCase.cpp:
* Source/JavaScriptCore/bytecode/InlineCacheCompiler.h:
* Source/JavaScriptCore/bytecode/PutByVariant.h:
* Source/JavaScriptCore/inspector/agents/InspectorDebuggerAgent.cpp:
* Source/JavaScriptCore/jit/ExecutableAllocator.cpp:
* Source/JavaScriptCore/parser/SourceTaintedOrigin.cpp:
* Source/JavaScriptCore/parser/SourceTaintedOrigin.h:
* Source/JavaScriptCore/runtime/FunctionConstructor.h:
* Source/WebCore/Modules/webaudio/OscillatorNode.h:
* Source/WebCore/Modules/webcodecs/WebCodecsVideoFrame.cpp:
* Source/WebCore/Modules/webtransport/DatagramSink.cpp:
* Source/WebCore/accessibility/AXObjectCache.cpp:
* Source/WebCore/css/CSSVariableReferenceValue.h:
* Source/WebCore/dom/LoadableClassicScript.cpp:
* Source/WebCore/dom/LoadableImportMap.cpp:
* Source/WebCore/dom/LoadableModuleScript.cpp:
* Source/WebCore/dom/PendingScript.cpp:
* Source/WebCore/html/HTMLDetailsElement.cpp:
* Source/WebCore/html/MediaDocument.cpp:
* Source/WebCore/html/parser/CSSPreloadScanner.cpp:
* Source/WebCore/inspector/agents/page/PageCanvasAgent.cpp:
* Source/WebCore/layout/floats/FloatingContext.cpp:
* Source/WebCore/layout/formattingContexts/block/BlockFormattingContext.cpp:
* Source/WebCore/layout/formattingContexts/block/BlockFormattingQuirks.cpp:
* Source/WebCore/layout/formattingContexts/inline/InlineContentCache.h:
* Source/WebCore/layout/formattingContexts/inline/ruby/RubyFormattingContext.cpp:
* Source/WebCore/layout/formattingContexts/inline/ruby/RubyFormattingContext.h:
* Source/WebCore/layout/formattingContexts/inline/text/TextUtil.cpp:
* Source/WebCore/layout/integration/LayoutIntegrationCoverage.h:
* Source/WebCore/layout/integration/inline/LayoutIntegrationInlineContent.cpp:
* Source/WebCore/layout/integration/inline/LayoutIntegrationInlineContentPainter.h:
* Source/WebCore/page/Location.h:
* Source/WebCore/page/OpportunisticTaskScheduler.cpp:
* Source/WebCore/page/PageConfiguration.cpp:
* Source/WebCore/page/RemoteDOMWindow.cpp:
* Source/WebCore/platform/AudioEncoder.cpp:
* Source/WebCore/platform/ThreadTimers.cpp:
* Source/WebCore/platform/adwaita/ScrollbarThemeAdwaita.cpp:
* Source/WebCore/platform/adwaita/ThemeAdwaita.cpp:
* Source/WebCore/platform/audio/HRTFElevation.cpp:
* Source/WebCore/platform/graphics/filters/software/FELightingSoftwareApplier.cpp:
* Source/WebCore/platform/graphics/texmap/GraphicsContextGLTextureMapperANGLE.cpp:
* Source/WebCore/platform/network/BlobPart.h:
* Source/WebCore/rendering/MotionPath.cpp:
* Source/WebCore/rendering/RenderAttachment.cpp:
* Source/WebCore/rendering/RenderThemeAdwaita.cpp:
* Source/WebCore/rendering/mathml/MathOperator.cpp:
* Source/WebCore/storage/StorageUtilities.h:
* Source/WebCore/workers/service/context/ServiceWorkerThreadProxy.cpp:
* Source/WebCore/workers/shared/SharedWorkerObjectConnection.cpp:
* Source/WebKit/GPUProcess/ShapeDetection/RemoteBarcodeDetector.cpp:
* Source/WebKit/GPUProcess/graphics/RemoteImageBuffer.cpp:
(RemoteImageBuffer::getPixelBuffer):
(RemoteImageBuffer::getShareableBitmap):
(RemoteImageBuffer::filteredNativeImage):
* Source/WebKit/GPUProcess/graphics/ScopedWebGLRenderingResourcesRequest.cpp:
* Source/WebKit/GPUProcess/media/RemoteAudioDestinationManager.h:
* Source/WebKit/Platform/IPC/Connection.cpp:
* Source/WebKit/UIProcess/BrowsingContextGroup.cpp:
* Source/WebKit/UIProcess/DisplayLinkProcessProxyClient.cpp:
* Source/WebKit/UIProcess/WebProcessProxy.cpp:
* Source/WebKit/WebProcess/GPU/graphics/RemoteImageBufferProxy.cpp:
* Source/WebKit/WebProcess/GPU/media/RemoteAudioDestinationProxy.cpp:

Canonical link: <a href="https://commits.webkit.org/268993@main">https://commits.webkit.org/268993@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3d636bce128e3a673a503e87426050283a2645a0

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/21191 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/21536 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/22245 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/23061 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/19685 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/21430 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/24805 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/21735 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/20915 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/21414 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/21106 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/18365 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/23913 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/18264 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/19197 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/25526 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/18437 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/19344 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/19398 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/23393 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/20583 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/19930 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/16941 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/24597 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/19220 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/5816 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/5091 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/23497 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/25864 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/19803 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/5641 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

rkirsling commented on Source/WebCore/platform/network/BlobPart.h:
```diff
@@ -27,6 +27,10 @@
 #include <wtf/URL.h>
 
+namespace IPC {
+template<typename T, typename> struct ArgumentCoder;
```
Out of curiosity, where did this one come from? (Also, you probably don't need the explicit `T`.)

donny-dont committed:
Non-unified build fixes, early October 2023 edition
https://bugs.webkit.org/show_bug.cgi?id=262719

Unreviewed non-unified build fix.

Run a non-unified build of Windows with clang-cl and fix any compilation
issues and warnings. Warnings were all around unused code.

* Source/JavaScriptCore/bytecode/AccessCase.cpp:
* Source/JavaScriptCore/bytecode/GetterSetterAccessCase.cpp:
* Source/JavaScriptCore/bytecode/InlineCacheCompiler.h:
* Source/JavaScriptCore/bytecode/PutByVariant.h:
* Source/JavaScriptCore/inspector/agents/InspectorDebuggerAgent.cpp:
* Source/JavaScriptCore/jit/ExecutableAllocator.cpp:
* Source/JavaScriptCore/parser/SourceTaintedOrigin.cpp:
* Source/JavaScriptCore/parser/SourceTaintedOrigin.h:
* Source/JavaScriptCore/runtime/FunctionConstructor.h:
* Source/WebCore/Modules/webaudio/OscillatorNode.h:
* Source/WebCore/Modules/webcodecs/WebCodecsVideoFrame.cpp:
* Source/WebCore/Modules/webtransport/DatagramSink.cpp:
* Source/WebCore/accessibility/AXObjectCache.cpp:
* Source/WebCore/css/CSSVariableReferenceValue.h:
* Source/WebCore/dom/LoadableClassicScript.cpp:
* Source/WebCore/dom/LoadableImportMap.cpp:
* Source/WebCore/dom/LoadableModuleScript.cpp:
* Source/WebCore/dom/PendingScript.cpp:
* Source/WebCore/html/HTMLDetailsElement.cpp:
* Source/WebCore/html/MediaDocument.cpp:
* Source/WebCore/html/parser/CSSPreloadScanner.cpp:
* Source/WebCore/inspector/agents/page/PageCanvasAgent.cpp:
* Source/WebCore/layout/floats/FloatingContext.cpp:
* Source/WebCore/layout/formattingContexts/block/BlockFormattingContext.cpp:
* Source/WebCore/layout/formattingContexts/block/BlockFormattingQuirks.cpp:
* Source/WebCore/layout/formattingContexts/inline/InlineContentCache.h:
* Source/WebCore/layout/formattingContexts/inline/ruby/RubyFormattingContext.cpp:
* Source/WebCore/layout/formattingContexts/inline/ruby/RubyFormattingContext.h:
* Source/WebCore/layout/formattingContexts/inline/text/TextUtil.cpp:
* Source/WebCore/layout/integration/LayoutIntegrationCoverage.h:
* Source/WebCore/layout/integration/inline/LayoutIntegrationInlineContent.cpp:
* Source/WebCore/layout/integration/inline/LayoutIntegrationInlineContentPainter.h:
* Source/WebCore/page/Location.h:
* Source/WebCore/page/OpportunisticTaskScheduler.cpp:
* Source/WebCore/page/PageConfiguration.cpp:
* Source/WebCore/page/RemoteDOMWindow.cpp:
* Source/WebCore/platform/AudioEncoder.cpp:
* Source/WebCore/platform/ThreadTimers.cpp:
* Source/WebCore/platform/adwaita/ScrollbarThemeAdwaita.cpp:
* Source/WebCore/platform/adwaita/ThemeAdwaita.cpp:
* Source/WebCore/platform/audio/HRTFElevation.cpp:
* Source/WebCore/platform/graphics/filters/software/FELightingSoftwareApplier.cpp:
* Source/WebCore/platform/graphics/texmap/GraphicsContextGLTextureMapperANGLE.cpp:
* Source/WebCore/platform/network/BlobPart.h:
* Source/WebCore/rendering/MotionPath.cpp:
* Source/WebCore/rendering/RenderAttachment.cpp:
* Source/WebCore/rendering/RenderThemeAdwaita.cpp:
* Source/WebCore/rendering/mathml/MathOperator.cpp:
* Source/WebCore/storage/StorageUtilities.h:
* Source/WebCore/workers/service/context/ServiceWorkerThreadProxy.cpp:
* Source/WebCore/workers/shared/SharedWorkerObjectConnection.cpp:
* Source/WebKit/GPUProcess/ShapeDetection/RemoteBarcodeDetector.cpp:
* Source/WebKit/GPUProcess/graphics/RemoteImageBuffer.cpp:
(RemoteImageBuffer::getPixelBuffer):
(RemoteImageBuffer::getShareableBitmap):
(RemoteImageBuffer::filteredNativeImage):
* Source/WebKit/GPUProcess/graphics/ScopedWebGLRenderingResourcesRequest.cpp:
* Source/WebKit/GPUProcess/media/RemoteAudioDestinationManager.h:
* Source/WebKit/Platform/IPC/Connection.cpp:
* Source/WebKit/UIProcess/BrowsingContextGroup.cpp:
* Source/WebKit/UIProcess/DisplayLinkProcessProxyClient.cpp:
* Source/WebKit/UIProcess/WebProcessProxy.cpp:
* Source/WebKit/WebProcess/GPU/graphics/RemoteImageBufferProxy.cpp:
* Source/WebKit/WebProcess/GPU/media/RemoteAudioDestinationProxy.cpp:

Canonical link: https://commits.webkit.org/268993@main
